src/Databases/DatabaseOrdinary.cpp

```diff
@@ -210,6 +271,36 @@ LoadTaskPtr DatabaseOrdinary::startupTableAsync(
                 /// until startup finished.
                 auto table_lock_holder = table->lockForShare(RWLockImpl::NO_QUERY, getContext()->getSettingsRef().lock_acquire_timeout);
                 table->startup();
+
+                /// If table is ReplicatedMergeTree after conversion from MergeTree,
+                /// it is in readonly mode due to metadata in zookeeper missing.
+                if (auto * rmt = table->as<StorageReplicatedMergeTree>())
+                {
+                    auto convert_to_replicated_flag_path = fs::path(getContext()->getPath()) / "data" / name.database / name.table / "flags" / "convert_to_replicated";
+                    if (fs::exists(convert_to_replicated_flag_path))
+                    {
+                        if (rmt->isTableReadOnly())
```

Comment:
It may be readonly for a different reason (although it's unlikely)

Reply:
Not sure what to do in such cases. Should we just ignore the exceptions from `restoreMetadataInZooKeeper` in case table is readonly not for missing metadata reason?

Reply:
Let's better check `has_metadata_in_zookeeper` instead of `is_readonly`